bitnami/nginx-ingress-controller/README.md

```diff
@@ -158,6 +158,7 @@ As an alternative, you can use of the preset configurations for pod affinity, po
 | `dhParam`                              | A base64ed Diffie-Hellman parameter                                                                                                                | `""`                                       |
 | `tcp`                                  | TCP service key:value pairs                                                                                                                        | `{}`                                       |
 | `udp`                                  | UDP service key:value pairs                                                                                                                        | `{}`                                       |
+| `portNamePrefix`                       | Prefix for TCP and UDP ports names in ingress controller service                                                                                   | `""`                                       |
```

Comment:
Both `tcp` and `udp` parameters allow you to use any prefix for the port names. For instance, using the values below as example:
```yaml
tcp:
  PREFIX-FOO: "5555"
udp:
  PREFIX-BAR: "6666"
```
The result is what you want, see:

```console
$ helm template nginx-ingress-controller oci://registry-1.docker.io/bitnamicharts/nginx-ingress-controller -f values.yaml -s templates/controller-service.yaml
(...)
spec:
  (...)
  ports:
    (...)
    - name: PREFIX-FOO-tcp
      port: PREFIX-FOO
      protocol: TCP
      targetPort: PREFIX-FOO-tcp
    - name: PREFIX-BAR-udp
      port: PREFIX-BAR
      protocol: UDP
      targetPort: PREFIX-BAR-udp

Reply:
There is a wrong port declaration in the example above, according to [ServicePort](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.31/#serviceport-v1-core) specification, it must be an integer.

One more example for tcp/udp `key` usage where we can't use arbitrary string is `containerPort` in [controller-deployment.yaml](https://github.com/bitnami/charts/blob/a8ceaef42ee34b57733f7d301441ff8f2e5cd349/bitnami/nginx-ingress-controller/templates/controller-deployment.yaml):

```yaml
    spec:
      containers:
        - name: controller
...
          ports:
...
            {{- range $key, $value := .Values.tcp }}
            - name: "{{ $key }}-tcp"
              containerPort: {{ $key }}
              protocol: TCP
            {{- end }}
            {{- range $key, $value := .Values.udp }}
            - name: "{{ $key }}-udp"
              containerPort: {{ $key }}
              protocol: UDP
            {{- end }}
```

It's not obvious from the values documentation, but `key` inside tcp/udp params must be a valid port number.

Reply:
That's clearly a bug. It should be:
```yaml
            {{- range $key, $value := .Values.tcp }}
            - name: "{{ $key }}-tcp"
              containerPort: {{ $value }}
              protocol: TCP
            {{- end }}
            {{- range $key, $value := .Values.udp }}
            - name: "{{ $key }}-udp"
              containerPort: {{ $value }}
              protocol: UDP
            {{- end }}
```

Reply:
I don't think so. `key` is an external port and `value` must be presented in the following format `<namespace/service name>:<service port/name>:[PROXY]:[PROXY]`. Please see https://kubernetes.github.io/ingress-nginx/user-guide/exposing-tcp-udp-services for more details.

Also I've found an example for tcp/udp params in https://github.com/bitnami/charts/pull/1396 (see [diff lines](https://github.com/bitnami/charts/pull/1396/files#diff-e0b453a4d4535124ed6f01a74890dc5cf5187e750a5bf00ed566ef814012c6beR514)):

```yaml
# TCP service key:value pairs
# Ref: https://github.com/kubernetes/contrib/tree/master/ingress/controllers/nginx/examples/tcp
##
tcp: {}
#  8080: "default/example-tcp-svc:9000"

# UDP service key:value pairs
# Ref: https://github.com/kubernetes/contrib/tree/master/ingress/controllers/nginx/examples/udp
##
udp: {}
#  53: "kube-system/kube-dns:53"
```

Reply:
> `key` is an external port and `value` must be presented in the following format `<namespace/service name>:<service port/name>:[PROXY]:[PROXY]`. Please see https://kubernetes.github.io/ingress-nginx/user-guide/exposing-tcp-udp-services for more details.

You're completely right, I wasn't familiar with this feature. I can confirm this is the expected format and what we use in the ConfigMap(s) below:

- https://github.com/bitnami/charts/blob/main/bitnami/nginx-ingress-controller/templates/tcp-configmap.yaml
- https://github.com/bitnami/charts/blob/main/bitnami/nginx-ingress-controller/templates/udp-configmap.yaml